src/pages/iou/ReceiptSelector/index.js

```diff
@@ -0,0 +1,200 @@
+        amount: PropTypes.number,
+        currency: PropTypes.string,
+        participants: PropTypes.arrayOf(
+            PropTypes.shape({
+                accountID: PropTypes.number,
+                login: PropTypes.string,
+                isPolicyExpenseChat: PropTypes.bool,
+                isOwnPolicyExpenseChat: PropTypes.bool,
+                selected: PropTypes.bool,
+            }),
+        ),
+    }),
+
+    /** Current user personal details */
+    currentUserPersonalDetails: personalDetailsPropType,
+
+    /** Used by drag and drop to determine if we have a file dragged over the view */
+    isDraggingOver: PropTypes.bool,
+};
+
+const defaultProps = {
+    receiptModal: {
+        isAttachmentInvalid: false,
+        attachmentInvalidReasonTitle: '',
+        attachmentInvalidReason: '',
+    },
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
+    report: {},
+    iou: {
+        id: '',
+        amount: 0,
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
+    ...withCurrentUserPersonalDetailsDefaultProps,
+    isDraggingOver: false,
+};
+
+function ReceiptSelector({route, currentUserPersonalDetails, isDraggingOver, iou, report, receiptModal}) {
+    const iouType = useRef(lodashGet(route, 'params.iouType', ''));
+    const reportID = useRef(lodashGet(route, 'params.reportID', ''));
+    const isAttachmentInvalid = lodashGet(receiptModal, 'isAttachmentInvalid', false);
+    const attachmentInvalidReasonTitle = lodashGet(receiptModal, 'attachmentInvalidReasonTitle', '');
+    const attachmentInvalidReason = lodashGet(receiptModal, 'attachmentInvalidReason', '');
+    const [receiptImageTopPosition, setReceiptImageTopPosition] = useState(0);
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const {translate} = useLocalize();
+
+    /**
+     * Close the confirm modal.
+     */
+    const closeConfirmModal = useCallback(() => {
+        Receipt.clearUploadReceiptError();
+    }, []);
+
+    const uploadText = () =>
+        isSmallScreenWidth ? (
+            <Text style={[styles.subTextReceiptUpload]}>
+                {translate('receipt.chooseReceiptBeforeEmail')}
+                <View style={{flexDirection: 'row'}}>
+                    <CopyTextToClipboard
+                        text="receipts@expensify.com"
+                        textStyles={[styles.textBlue]}
+                    />
+                </View>
+            </Text>
+        ) : (
+            <Text style={[styles.subTextReceiptUpload]}>
+                {translate('receipt.dragReceiptBeforeEmail')}
+                <View style={{flexDirection: 'row'}}>
+                    <CopyTextToClipboard
+                        text="receipts@expensify.com"
+                        textStyles={[styles.textBlue]}
+                    />
+                </View>
+                {translate('receipt.dragReceiptAfterEmail')}
+            </Text>
+        );
```

Comment:
Why is this logic based on screenwidth?

Reply:
We want a different style based on web vs mobile web.

Reply:
The only difference here is `receipt.dragReceiptAfterEmail` which is `or choose a file to upload below.` On mWeb we do have a button to choose a file. But if we don't need that text, we should only make the if condition for this extra text and not the whole view.